engine/message/db.go

```diff
@@ -360,21 +353,40 @@ func NewDB(ctx context.Context, db *sql.DB, c *Config, a alertlog.Store) (*DB, e
 			left join user_contact_methods cm on cm.id = msg.contact_method_id
 			left join notification_channels chan on chan.id = msg.channel_id
 			where
-				sent_at > now() - '10 minutes'::interval or
+				sent_at >= $1 or
 				last_status = 'pending' and
 				(msg.contact_method_id isnull or msg.message_type = 'verification_message' or not cm.disabled)
 		`),
 	}, p.Err
 }
 
+var maxMessageAge = maxThrottleDuration(perCMThrottle, globalCMThrottle)
+
 func (db *DB) currentQueue(ctx context.Context, tx *sql.Tx, now time.Time) (*queue, error) {
-	rows, err := tx.Stmt(db.messages).QueryContext(ctx)
+	cutoff := now.Add(-maxMessageAge)
+	sentSince := db.lastSent
+	if sentSince.IsZero() {
+		sentSince = cutoff
+	}
+
+	// TODO: possibly dedup (e.g. multiple engines)
```

Comment:
This would be a net-new issue, so should be addressed here. IIRC it needs some attention to verify that we're not going to get the same message multiple times in `sentMessages` -- otherwise it would affect the limits.

The main issue is we are scanning for `sent_at >= <cutoff>`. If that returns a message sent after the transaction's `now` time we do want to include it, but the next tick will result in reading the same message again.

It may not be possible in practice right now, as the message module should operate exclusively. However, doing some deduplication via a map would ensure the counts are always accurate despite future changes.

Reply:
this should now be resolved